### ISSUE TITLE
Code for power incorrect in GUIPatternPassword incorrect for powers > 2 and < 0

### ISSUE THREAD

**Issue by midopa** [TeamWin/Team-Win-Recovery-Project#646]
Code in question: https://github.com/TeamWin/Team-Win-Recovery-Project/blob/7a48a664fd00a188044da348f34cc2e800b8a3fe/gui/patternpassword.cpp#L233

If you try to compute 2^3, you will get 16 != 2^3 = 8.

This is because of the cumulative in-place multiplication going on in the while loop.

Iterations:
- 2 *= 2 -> 4 -> OK for 2^2
- 4 *= 4 -> 16 -> Bug

Also, this code won't work for powers that are less than 0, like 2^-1.

But it seems this code was written solely to calculate powers of two. I suggest replacing it with a 'squared' static inline function or macro.

**Comment by that1** [comment]
Yes, an inline function for "squared" is definitely better than a broken "pow" function.

**Comment by nailyk-fr** [comment]
Is it a way to do: 

```
        static int pow(int x, int i)
        {
                int result = 1;  
                if (i<0)
                        return 0;
                while(i-- > 0)
                        result *= x;
                return result;
        }
```

? 

Comparison between "math.h" pow and this pow functions: 

> ./pow_test 2 2
> math_pow: 4, twrp_pow: 4
> ./pow_test 2 0
> math_pow: 1, twrp_pow: 1
> ./pow_test 2 -9
> math_pow: 0, twrp_pow: 0
> ./pow_test 5 5
> math_pow: 3125, twrp_pow: 3125
> ./pow_test 8 8
> math_pow: 16777216, twrp_pow: 16777216

(It was tested on my 64bits computer which return INT_MAX 2147483647, not on the phone )
